app/models/search.rb

```diff
@@ -73,4 +54,24 @@ def validate_dates
       errors.delete(:keywords)
     end
   end
+
+  def build_date(value, field)
+    year = value[1]
+    month = value[2]
+    day = value[3]
+
+    if field == "from"
```

Comment:
Nice idea to make this into a method. 
You could avoid the if condition by returning the output and assigning when you call the method. 
e.g.
```
@date_from = build_date(value)
```

Reply:
Great! thanks for that.  Implemented as per your suggestion.